lib/spack/spack/relocate.py

```diff
@@ -44,6 +46,64 @@ def get_existing_rpath(path_name, patchelf_executable):
         tty.die('relocation not supported for this platform')
     return retval
 
+def change_dylib(path_name, rpaths):
+    """
+    Return the RPATHS in given file as a list of strings.
+    """
+    command = which('otool')
+    output  = command("-l", path_name, output=str, err=str)
+    if command.returncode != 0:
+        tty.warn('failed reading rpath for %s.' % path_name)
+        return False
+    last_cmd = None
+    path = ''
+    deps = []
+    for line in output.split('\n'):
```

Comment:
Could you give an example of what the regular expressions are looking for? And wouldn't much simpler code that checks whether things start with the orig_path? You don't need complex code for that.